languagetool-core/src/main/java/org/languagetool/rules/AbstractCheckCaseRule.java

```diff
@@ -104,7 +106,7 @@ public RuleMatch[] match(AnalyzedSentence sentence) {
           }
           ruleMatch = new RuleMatch(this, sentence, startPos, endPos, msg, getShort());
           if (subRuleSpecificIds) {
-            ruleMatch.setSpecificRuleId(StringTools.toId(getId() + "_" + correctPhrase));
+            ruleMatch.setSpecificRuleId(StringTools.toId(getId() + "_" + correctPhrase, language.getShortCode()));
```

Comment:
Could you use `language` instead of its code? This leads to better type-safety.

Reply:
Done!